WCFDataService/Client/System/Data/Services/Client/MaterializeFromAtom.cs

```diff
@@ -69,6 +69,11 @@ internal class MaterializeAtom : IDisposable, IEnumerable, IEnumerator
         private bool moved;
 
         /// <summary>
+        /// The ODataResponseMessage to dispose when we are done.
+        /// </summary>
+        private IODataResponseMessage responseMessage;
```

Comment:
Acually the same is true for GET. If you do ctx.Execute by hand - and don't do any enumeration/foreach of the result - the message will not be disposed. This has actually always been the case, but the difference is now the stream (the stuff that we are actually interested in disposing) is a http/network stream whereas before it was a memory stream. So the issue have always been there, the diference is just what type of stream we are talking about. At some point GC will handle these scenarios.

Is this correctly understood?
Is it a major issue due to the fact that the stream now is a Network stream? If so, I really do not see any way around this, except we could make this a configuration/flag that you need to enable to set to get this new optimized behaviour, but at the expense of you/the programmer have to ensure they enumerate the results or dispose them.